prow/pod-utils/decorate/podspec.go

```diff
@@ -60,6 +60,10 @@ func ProwJobToPod(pj kube.ProwJob, buildID string) (*v1.Pod, error) {
 	podLabels[kube.CreatedByProw] = "true"
 	podLabels[kube.ProwJobTypeLabel] = string(pj.Spec.Type)
 	return &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
@deads2k @sttts expectedly, `yaml.Marshal` on a `v1.Pod` without a literal `TypeMeta` ends us with a non-viable YAML. What's the more correct way to serialize? Register to the scheme, black magic, ???, profit? Does it matter?

Reply:
> What's the more correct way to serialize? Register to the scheme, black magic, ???, profit? Does it matter?

We use `runtime.Encode`.  The same type can be used to serialize to different target GroupVersions, so choosing one doesn't make sense for the apiserver scheme (which this re-uses).  I think you can also do a .SetObjectKind.

Reply:
Is there an example somewhere? Getting stuck here

Reply:
pod.GetObjectKind.SetGroupVersionKind